cmd/kubeadm/app/cmd/phases/upgrade/node/preflight.go

```diff
@@ -67,6 +67,10 @@ func runPreflight(c workflow.RunData) error {
 		initConfig.NodeRegistration.ImagePullSerial = data.Cfg().Node.ImagePullSerial
 
 		if !data.DryRun() {
+			if err := preflight.RunImagesExistCheck(utilsexec.New(), initConfig, data.IgnorePreflightErrors()); err == nil {
+				fmt.Println("[upgrade/preflight] All images required for setting up a Kubernetes cluster exist")
+				return nil
+			}
 			fmt.Println("[upgrade/preflight] Pulling images required for setting up a Kubernetes cluster")
 			fmt.Println("[upgrade/preflight] This might take a minute or two, depending on the speed of your internet connection")
```

Comment:
this line can be removed, but it helps newbie users who anticipate a progress but are on slow internet.

Reply:
It's a good line, since it can take several minutes - with all the progress output staying in the runtime.